### ISSUE TITLE
autoHeight breaks on Safari

### ISSUE THREAD

**Issue by ndimatteo** [OwlCarousel2/OwlCarousel2#556]
When OwlCarousel initializes on safari the height is set to 1px for some reason. This seems to only affect carousels with the 'autoHeight' plugin being used. The only way to jig it properly is either going to a new slide (recalculates height properly) or refreshing the page (works sometimes).

This only happens in safari (both desktop & mobile iOS)

You can see an example of it happening here: http://dev.yarmouth.nickdimatteo.com/listing/633-maryland-ave-ne/

and here's a screenshot:

| ERROR | CORRECT |
| :-: | :-: |
| ![error](http://f.cl.ly/items/2u3T1r443H3c323B3C1J/Screen%20Shot%202014-10-14%20at%203.47.07%20PM.png) | ![correct](http://cl.ly/image/1E3k1n3M0F1l/Screen%20Shot%202014-10-14%20at%203.53.21%20PM.png) |

**Comment by joeyjoejoejr** [comment]
Can someone verify that this is still an issue?

**Comment by seamofreality** [comment]
Yes, currently working on a project where this is still an issue.

**Comment by lolsps** [comment]
Still an issue. I decided to avoid this option and make images the same.

**Comment by fulopattila122** [comment]
As of today it's still an issue and neither @ndimatteo's nor @unostella's workaround did not work for me. In case of the latter it says `Uncaught ReferenceError: owl is not defined`. Removing the autoHeight option did help.

**Comment by subduer** [comment]
Hi! I have same problem: Uncaught ReferenceError: owl is not defined.
It appear only once, after web browser starts. Refresh web or navigate somewhere on web makes my OWL working.
None of these workarounds works :(  Any ideas? Any other workarounds?

**Comment by Haprog** [comment]
This issue still seems to persist but also affects Chrome and other browsers.

And I'm pretty sure these issues are all duplicates of this one: #760, #835, #1381, #2039, #2054, #2113.

I'm dynamically creating and initialising the carousel after page has been loaded (window load event has been executed already) and had to figure out a workaround for this problem. This is what I came up with:

Just add this `onload` handler to each `<img>` in the carousel:
```html
<img src="..." alt="" onload="if ($(this).closest('.owl-height').height() === 1) { $(this).closest('.owl-carousel').trigger('refresh.owl.carousel'); }">
```

This seems to work nicely. It will trigger a refresh of the carousel every time an image in the carousel has loaded if the carousel's auto height wrapper's height is still 1px (initial size).

First I tried a simpler approach of always just triggering the refresh on image load event but that seemed to cause an infinite loop of triggering the load events and carousel refreshes and the browser would cause constant high CPU usage when the page is open. So this is why I had to add the check to only refresh if the height hasn't been set yet.

**Comment by Haprog** [comment]
Actually my workaround above doesn't seem to be foolproof. It is better than nothing, but still in some cases it seems to not set the height. When I was testing in our project without the fix above I rarely saw the height problem normally, but I could consistently reproduce it when using Chrome's network throttling set to Fast 3G. With the above fix it seems to work ok on Fast 3G, but still fails to set the height on Slow 3G throttling.

**Comment by robertrhu** [comment]
I was having this issue due to slow site load. Came up with the following solution...
https://github.com/OwlCarousel2/OwlCarousel2/issues/2185

**Comment by pascalporedda** [comment]
It took a while, though I've merged a PR proposing a change related to this issue. Have been testing this out with a slow network connection and wasn't able to reproduce the 1px behaviour.

The fix will be in the next release. If you still encounter this issue feel free to reply to me.

**Comment by gleenk** [comment]
This bug is still present on 2.2.1 version but solved on 2.3.4
